app/models/allocation.rb

```diff
@@ -2,4 +2,37 @@ class Allocation < ActiveRecord::Base
   belongs_to :country
   belongs_to :protected_area
   belongs_to :user
+
+  def self.to_csv
+    csv = ''
+    allocation_columns = Allocation.new.attributes.keys
+    allocation_columns.delete("created_at")
+    allocation_columns.delete("updated_at")
+    allocation_columns.delete("country_id")
+    allocation_columns << "Country"
+
+    allocation_columns.map! { |e|
+      e.gsub("_", " ").capitalize
+    }
+
+    csv << allocation_columns.join(',')
+    csv << "\n"
+
+    allocations = Allocation.all
+
+    allocations.to_a.each do |allocation|
+      allocation_attributes = allocation.attributes
+      allocation_attributes.delete("created_at")
+      allocation_attributes.delete("updated_at")
+      allocation_attributes[:country] = allocation.country.iso3
+      allocation_attributes.delete("country_id")
```

Comment:
Same as above. Or you could use something like:
```
allocations.each do |allocation|
  allocation_attributes = allocation_columns.map { |c| allocation.send(c) }
  ...
  #then you need to overwrite the country attribute as you are already doing plus any extra formatting
end
```
Note that `.to_a` is not needed

Reply:
Nice, thanks @Levia